flamedisx/lxe_blocks/quanta_generation.py

```diff
@@ -15,7 +15,7 @@
 class MakeERQuanta(fd.Block):
 
     dimensions = ('quanta_produced', 'energy')
-    extra_dimensions = (('quanta_produced_noStep', False),)
+    bonus_dimensions = (('quanta_produced_noStep', False),)
```

Comment:
Can you explain what is happening here? I.e. did you rename extra_dimensions to bonus_dimensions, does this relate to our earlier conversations about this attribute (in https://github.com/FlamTeam/flamedisx/pull/127) and the way the attributes are described in source.py (which autofeeds the docs)? Probably it's also related to returning an extra `quanta_produced` domain from `domain_dict_bonus`,

Reply:
This is just a simplification of the change made way back in #127, yes. Back then I had `extra_dimensions` and `bonus_dimensions` (the distinction there was only made to predict what I'd do in this PR, and I ended up simplifying things when it came to coding it up). The user must handle the dimsize/stepping calculation manually for any of these in `_calculate_dimsizes_special()` and tensor construction for them must be manually done in `_domain_dict_bonus()`. This currently happens here, and in `quanta_splitting.py` in the NEST sources. The flag just flags up whether or not the dimension represents a hidden variable that will be summed over in the final computation, and thus whether a corresponding variable stepping scaling should be applied or not. I've changed the comment a little in `block_surce.py` to make that clearer.